### ISSUE TITLE
how to use mmrazor1.x 

### ISSUE THREAD

**Issue by wwjwy** [open-mmlab/mmrazor#367]
Based on my own dataset, I trained a model using mmseg1.x. Now I want to realize knowledge distillation, resnet101+deeplab--->resnet18+deeplab. 
I configured the mmrazor environment, but in the distill config file https://github.com/open-mmlab/mmrazor/blob/bbb58f1a5c2fe2878484856767dba540092bc7bf/configs/distill /mmseg/cwd/cwd_logits_pspnet_r101-d8_pspnet_r18-d8_4xb2-80k_cityscapes-512x1024.py#L2
I don't know where I can modify the default settings of mmseg, such as the number of categories to be segmented, data set selection, etc.? 
Can you give me some help?

**Comment by TinyTigerPan** [comment]
The config in MMRazor only define the distillation configuration. If you want modify settings of MMSeg, you can modify config in MMSeg, then import them to MMRazor config by `mmseg::_base_/datasets/cityscapes_modify.py` in the front.
As for how to change MMSeg config, you can refer [mmseg doc](https://mmsegmentation.readthedocs.io/en/1.x/user_guides/1_config.html)

**Comment by wwjwy** [comment]
> The config in MMRazor only define the distillation configuration. If you want modify settings of MMSeg, you can modify config in MMSeg, then import them to MMRazor config by `mmseg::_base_/datasets/cityscapes_modify.py` in the front. As for how to change MMSeg config, you can refer [mmseg doc](https://mmsegmentation.readthedocs.io/en/1.x/user_guides/1_config.html)

Thank you for your help! it runs well now.

**Comment by wwjwy** [comment]
When i distill transformer seg models --->CNN seg model, there is something wrong. 
assert preds_S.shape[-2:] == preds_T.shape[-2:]
The transformer and CNN has different output shape, can you show me how to modify the config file?

**Comment by TinyTigerPan** [comment]
`assert preds_S.shape[-2:] == preds_T.shape[-2:]` used to make sure feature shape between teacher and student are the same, to make sure the KD loss can be calculate. For example, `MSE(Tensor shape [2, 3, 4, 4], Tensor shape [2, 3, 5, 5])` can not get a result. As far as I know, many Transformer based backbone like PVT, Swin, ..., their backbone output is also with shape [B, C, H, W], it can used to distill with CNN based backbone. 
So, my advice is 
1) check your `preds_S.shape` and `preds_T.shape`, make sure `preds_S.shape[-2:] == preds_T.shape[-2:]`, if not, check your config, make sure your `recorders` get right feature.
2) if shape between teacher and student are different, but you still want to distill them, you can align dimension by using `connector` in MMRazor, [example config](https://github.com/open-mmlab/mmrazor/blob/1.x/configs/distill/mmcls/ofd/ofd_backbone_resnet50_resnet18_8xb16_cifar10.py#L43)

**Comment by wwjwy** [comment]
Thank you for your help!

**Comment by wwjwy** [comment]
After training the cwd model, i run the tools/test.py, but i can only get evaluation metrics. How can i get the segmentation images predicted by the distilled student model?

**Comment by TinyTigerPan** [comment]
MMRazor `tools/test.py` seems not provide `--show`, but MMSeg provide, like [this](https://mmsegmentation.readthedocs.io/en/latest/inference.html#test-a-dataset). You can try use MMSeg `tools/test.py` with `--show`, or if your server has no GUI, you can save the image to disk by `--show-dir`.

**Comment by wwjwy** [comment]
when i use MMSeg tools/test.py with the pth file obtained in mmrazor, there is some issues.
The model and loaded state dict do not match exactly

unexpected key in source state_dict: architecture.backbone.conv1.weight, architecture.backbone.bn1.weight, architecture.backbone.bn1.bias, architecture.backbone.bn1.running_mean, architecture.backbone.bn1.running_var, architecture.backbone.bn1.num_batches_tracked, architecture.backbone.layer1.0.conv1.weight, architecture.backbone...................

**Comment by wwjwy** [comment]
Thanks

**Comment by wwjwy** [comment]
Thank you very much for your help!
